staging/src/k8s.io/component-base/metrics/prometheus/restclient/metrics.go

```diff
@@ -54,13 +54,36 @@ var (
 		},
 		[]string{"code", "method", "host"},
 	)
+
+	certExpiration = k8smetrics.NewGauge(
+		&k8smetrics.GaugeOpts{
+			Name: "rest_client_certificate_expiration_seconds",
+			Help: "Gauge of the lifetime of the client certificate. The value is the date the certificate will expire in seconds since January 1, 1970 UTC.",
```

Comment:
This is mis-leading because it's only reported for certs from exec plugin.
Either move reporting to higher-level client code that catches *all* certs (including those passed via flags or created via bootstrap-kubeconfig) or explicitly state that these metrics are only for exec plugin certs

Reply:
Updated the name of both metrics to contain "exec plugin". Let me know if you think its any better.